pkg/sentry/identity/kubernetes/validator.go

```diff
@@ -54,7 +54,14 @@ func (v *validator) Validate(id, token string) error {
 
 	podSa := prts[3]
 	podNs := prts[2]
-	if id != fmt.Sprintf("%s:%s", podSa, podNs) {
+
+	if namespace != "" {
+		if podNs != namespace {
+			return errors.Errorf("%s: namespace mismatch. received namespace: %s", errPrefix, namespace)
+		}
+	}
+
+	if id != fmt.Sprintf("%s:%s", podNs, podSa) {
```

Comment:
the previous code is `podSa:podNs`. isn't it breaking change?

Reply:
No, the sidecar injector was changed accordingly to inject this new format.